### DISCUSSION TITLE
How to add capacitor with newly build quasar app?

### DISCUSSION THREAD

**Discussion by KrishnKumar-95** [quasarframework/quasar#16261] [CLI - Capacitor / Cordova modes] [answered]
I created quasar app with below steps but confused where and how the capacitor will be used.
1. Download Quasar CLI : npm i -g @quasar/cli
2. Initialize Quasar App : npm init quasar
3. ✔ What would you like to build? › App with Quasar CLI, let's go!
4. ✔ Project folder: … driver
5. ✔ Pick Quasar version: › Quasar v2 (Vue 3 | latest and greatest)
6. ✔ Pick script type: › Typescript
7. ✔ Pick Quasar App CLI variant: › Quasar App CLI with Vite
8. ✔ Package name: … driver
9. ✔ Project product name: (must start with letter if building mobile apps) … Driver
10. ✔ Project description: … Taxigo driver app providing business to drivers
11. ✔ Author: … Krishan Kumar
12. ✔ Pick a Vue component style: › Composition API with <script setup>
13. ✔ Pick your CSS preprocessor: › Sass with SCSS syntax
14. ✔ Check the features needed for your project: › ESLint, State Management (Pinia), Axios, Vue-i18n
15. ✔ Pick an ESLint preset: › Prettier
16. ✔ Install project dependencies? (recommended) › No, I will handle that myself
17. cd taxigo-driver
18. yarn
19. yarn lint --fix
20. quasar dev

**Comment by jimoquinn** [comment, marked as answer]
I know this is an old question, but having just started working with Quasar/Capacitor Android myself, the answers are fresh in my mind.  In the current set of docs for Quasar under Developing Mobile Apps (either Vite or Webpack) there you'll find a set of instructions on how to get Capacitor added to your project.  

The actual commands run from your project directory are `quasar mode add capacitor` and to build, use `quasar dev -m capacitor -T [android|ios]`, but there is a good bit of upfront work required before you can build the app.  You have to install the Android Studio and pick the build targets (SDKs) or if on a Mac and targeting iOS, install XCode, and in both cases setup a handful of environment variables.  Not really difficult, but does take a little with the download and the these apps do a bit of self configuration.  

It may look daunting at first, but Quasar's docs are so good that if you follow them, you shouldn't have any issues building your first app and deploying it to the phone emulator.

**Reply by Max-Kucher** [reply]
Bro, but how can we build the app "not for emulator"?

I compiled the .apk file and installed it on my phone. The app opened and tried to connect to the quasar dev server (which is not available because it is only available for PC), I got a black screen.

In this regard, I have two questions:
1. How can I make a quasar production server instead of a dev server (there should be a difference)?
2. For an app with a counter, no server is needed at all, how can I build it so that it does not try to go online at all? There is only client logic in this app.

In general, tell me what to read or point me to a video with a detailed analysis of such issues. Thanks in advance.

**Comment by tenace2** [comment]
Well, the answer does have a lot of great information that was probably appropriate when it was posted, but for anybody else reviewing this note as of Feb 2025 and struggling with how to get a Quasar-Capacitor project scaffolded, here's an updated tip: 
 The quasar mode add capacitor command will default in v6.
 But subsequent installs of capacitor plugins will default in v7.
    ...the command "npm install @capacitor/android" will default in v7
       ...and that will conflict with the core v6 that Quasar defaulted in.
          ...so, to make the version dependencies work, you have to run npm install @capacitor/android@6.0.0

Here's a link to a blog I wrote on the subject:
https://tenace2.github.io/QuasarInstallGuide3/quasarscaffold/quasarscaffold.html

**Reply by developer-mtcllp** [reply]
This version related issue is resolved, now by default quasar provides capacitor v7 support
